### ISSUE TITLE
Bug with Segmentation fault

### ISSUE THREAD

**Issue by DoubleXKing** [cvlab-dresden/DSAC#2]
When I try to run ../../core/build/train_obj -oscript train_obj.lua
It post the following error:
Successfully loaded sensor transformation:
[0.9998282978278875, 0.008186003849805841, 0.01662420535123559, -0.01393324413905143;
  -0.008090415588790156, 0.9999503986429169, -0.005809081637597863, 0.05228905046770047;
  -0.01667093393273896, 0.005673587475537798, 0.9998449331606215, 0.02712006871814571;
  0, 0, 0, 1]

Parsing config file: default.config
focal length: 525
x shift: 0
y shift: 0
raw data (rescale rgb): 1
secondary focal length: 585
raw x shift: 0
raw y shift: 0
image width: 640
image height: 480
ransac iterations: 256
ransac iterations (refinement): 8
ransac batch size: 100
ransac refinement gradient sub sampling: 0.01
ransac inlier threshold: 10
ransac inlier threshold: 100
random draw: 1
object script: train_obj.lua

Loading training set ...
Reading file names... 
TORCH: Creating network.
TORCH: Set model to evaluation mode.
Training CNN.
Starting training round 1
TORCH: Set model to training mode.
Segmentation fault

I think my previous work are correct, but I am not sure what the problem of this Segmentation fault.
Could you please kindly give me some ideas?
I am appreciate for your help.

**Comment by ebrach** [comment]
Hello, my best guess is that there is something wrong with your data set. Assuming you are using the 7-Scenes data set, have you added all images and pose files to the appropriate folders (see Sec. 3 of the documentation)? You can check with the following line whether everything is at the correct place: Add to train_obj.cpp at line 228:

std::cout << "Data Set Size: " << trainDataset.size() << std::endl;

It should print some data set size > 0.

**Comment by DoubleXKing** [comment]
My data set size is 0.
Thanks for your help.

**Comment by ebrach** [comment]
You have to copy the 7-Scenes image and pose files into the following folder structure (just unzipping it will not work)

core
7scenes
-- 7scenes_chess
---- training
------ scene
-------- rgb_noseg -> move training color frames here, e.g. frame-000000.color.png
-------- depth_noseg -> move training depth frames here, e.g. frame-000000.depth.png
-------- poses -> move training pose files here, e.g. frame-000000.pose.txt
---- test -> same structure as training

You can also have a look at dataset.h which manages reading of data. 

Hope that helps,
Eric

**Comment by DoubleXKing** [comment]
Thanks. It works, I will continue to research your work
